pkg/asset/ignition/provisioningtoken.go

```diff
@@ -0,0 +1,46 @@
+package ignition
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+// tokenLen is how many bytes of random input go into generating the token.
+// It should be a multiple of 3 to render nicely in base64
+// encoding.  The current default is long; it's a lot of entropy.  The MCS
+// will have mitigations against brute forcing.
+const tokenLen = 30
+
+// ProvisioningToken implements https://github.com/openshift/enhancements/pull/443/
+type ProvisioningToken struct {
+	Token string
+}
+
+var _ asset.Asset = (*ProvisioningToken)(nil)
+
+// Dependencies returns no dependencies.
+func (a *ProvisioningToken) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate the token
+func (a *ProvisioningToken) Generate(asset.Parents) error {
+	b := make([]byte, tokenLen)
+	n, err := rand.Read(b)
+	if err != nil {
+		return err
+	}
+	if n != tokenLen {
```

Comment:
This check is not necessary. It is already guaranteed by the `rand.Read` function.

Reply:
Yeah, excess of paranoia there.  Removed.